### PR TITLE
Fix accessibility issues with error messages 

### PR DESCRIPTION
**Pull request by fitore-cds**
# Summary | Résumé

In this PR, there are two fixes for validation accessibility:
1. When errors appear at the top of the form, the focus will shift to that error container
2. When clicking on an error link in this container (error alert box), you’ll be “scrolledIntoView” of the label that sits above the input field and the field will have focus on it (previously it would scroll directly into the input field, so the label text couldn't be seen)

*Note: open to future improvements, if we find a way to not use JS

### CODE REVIEW

File: components/forms/Form/Form.tsx [hunk 1]
```diff
@@ -19,18 +19,32 @@ const InnerForm = (props: InnerFormProps & FormikProps<FormValues>) => {
...
     <>
       {formStatus ? <Alert type="error" heading={formStatus} /> : null}
       {errorList ? (
-        <Alert type="error" heading={t("input-validation.heading")} validation={true}>
+        <Alert
+          type="error"
+          heading={t("input-validation.heading")}
+          validation={true}
+          id={errorId}
+          tabIndex={0}
```
Comment (brdunfield) [inline components/forms/Form/Form.tsx:46]: Tab index! 👍 Great idea!

Comment (fitore-cds) [reply components/forms/Form/Form.tsx:46]: :-)

File: components/forms/Form/Form.tsx [hunk 2]
```diff
@@ -19,18 +19,32 @@ const InnerForm = (props: InnerFormProps & FormikProps<FormValues>) => {
...
 
   const errorList = props.errors ? getErrorList(props) : null;
+  const errorId = "gc-form-errors";
   const formStatus = props.status === "Error" ? t("server-error") : null;
   if (formStatus) {
-    window.scrollTo({
-      top: 0,
-      behavior: "smooth",
-    });
+    window.scrollTo({ top: 0, behavior: "smooth" });
```
Comment (brdunfield) [inline components/forms/Form/Form.tsx:25]: It's a minor issue and doesn't affect functionality, but the "smooth" scroll doesn't appear to be working anymore.

Comment (brdunfield) [reply components/forms/Form/Form.tsx:25]: it's probably being written by the useEffect below - in which case we can probably just remove this?

File: lib/validation.tsx [hunk 3]
```diff
@@ -10,6 +10,71 @@ import {
...
+ * @param e The click event of the error link
+ * @param id The id of the input field that has the error and we need to focus
+ */
+const scrollErrorInView = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>, id: string) => {
+  e.preventDefault();
+  const labelElement = document.getElementById(`label-${id}`);
+  const inputElement = document.getElementById(id);
+  if (labelElement && inputElement) {
+    inputElement.focus();
+    labelElement.scrollIntoView();
```
Comment (brdunfield) [inline lib/validation.tsx:74]: 👍 👍

**Review by brdunfield — APPROVED** [review]
This is really cool, and I didn't know about useEffect! Happy to learn about this.

LGTM! 🦑 

The only possible change is looking at that window.scrollTo in Form.tsx that's no longer smooth scrolling, and see if it can be removed.

File: components/forms/Form/Form.tsx (hunk 2 above)
Comment (fitore-cds) [reply components/forms/Form/Form.tsx]: Good catch, this one pertains to server-side errors from the submission lambda (either a notify error, or connection etc). But, even though it's not part of the validation, you're completely right that it should have the same accessibility fix applied to it, within the useEffect. 

Updated the branch, thanks for your help!

**Pull request merged**